server/index.js

```diff
@@ -29,6 +29,29 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId);
+  if (isNaN(productId) || productId < 0) {
+    res.status(400).json({ error: 'productId must be a positive integer' });
```

Comment:
You can pass a `new ClientError` to `next()` here, including your message and status code.